cli/npm/managed/resolvers/local.rs

```diff
@@ -1048,10 +1048,13 @@ fn symlink_package_dir(
   // need to delete the previous symlink before creating a new one
   let _ignore = fs::remove_dir_all(new_path);
 
+  let old_path = crate::util::path::relative_path(new_parent, old_path)
+    .unwrap_or_else(|| old_path.to_path_buf());
+
   #[cfg(windows)]
   return junction_or_symlink_dir(old_path, new_path);
```

Comment:
Just remembered that junctions can't be relative. We might need to update this code to handle that.

Reply:
@piscisaureus recommends attempting to do a symlink first, then if an EPERM occurs to use junctions from then on.

Reply:
I guess they'll just have to be absolute on windows then, that's annoying